docs/glossary.rst

```diff
@@ -0,0 +1,33 @@
+Glossary
```

Comment:
Does this file really belong to this PR?

Reply:
It was kinda "created" while writing the documentation. It is not 100% right here, but I figured "eh". Perfect is the enemy of good.